### DISCUSSION TITLE
Struggling with syntax in for eglot-workspace-configuration in .dir-locals.el

### DISCUSSION THREAD

**Discussion by mishok13** [joaotavora/eglot#1425] [Q&A] [answered]
I have a somewhat weird issue with Terraform's LSP server, which I was trying to address by changing the `workspaceFolders` setting. I just need to provide Eglot with a configuration in `.dir-locals.el` so that `M-x eglot-show-workspace-configuration` produces this:

```json
{
  "terraform-ls": {
    "workspaceFolders": [
      {
        "uri": "foo",
        "name": "bar"
      }
    ]
  }
}
```

Turns out, I'm hitting the limits of my ELisp knowledge here, as this is the closest I could get without `eglot` swearing at me:

```elisp
((nil
  . ((eglot-workspace-configuration
      . (terraform-ls
         (workspaceFolders
          (uri "foo" name "bar")))))))
```

which in turn produces this which is not correct (`workspaceFolders` is supposed to be a list): 

```json
{
  "terraform-ls": {
    "workspaceFolders": {
      "uri": "foo",
      "name": "bar"
    }
  }
}
```

So while the question is not exactly Eglot-specific, I'm still wondering if someone here with much stronger ELisp knowledge could give a helping hand. Thanks!

**Comment by mishok13** [comment, marked as answer]
Well, obviously writing this question led me on a path of discovery of correct syntax:
```elisp
((nil
  . ((eglot-workspace-configuration
      . (terraform-ls
         (workspaceFolders [(uri "foo" name "bar")]))))))
```

Apologies for spamming the discussions section. I'll show myself out.

**Reply by joaotavora** [reply]
Glad you fixed it.  Did you find this through trial and error or via the doc? I recommend using keyword symbols in the examples, but it's not necessary.  See https://joaotavora.github.io/eglot/#Examples

**Reply by mishok13** [reply]
Even though I recall seeing the examples, I did manage to miss this specific example which would save me a couple of hours at least https://joaotavora.github.io/eglot/#JSONRPC-objects-in-Elisp-1 I ended up with trial and error method, btw.
